qcodes/math/field_vector.py

```diff
@@ -250,30 +258,127 @@ def is_equal(self, other):
 
         return True
 
+    def __getitem__(self, component):
+        return self.get_components(component)[0]
+
+    def __setitem__(self, component, value):
+        self.set_component(**{component: value})
+
+    def __mul__(self, other):
+        if not isinstance(other, (float, int)):
```

Comment:
Try the following: 
```python
>>> import numpy as np 
>>> a = np.int32(2)
>>> isinstance(a, int) 
False 
```

Reply:
We have a similar situation going on in qcodes/dataset/sqlite_base.py on line 319 and 327. I propose to make a module in math called `datatypes.py` with the content

```python
import numpy as np 

int_type = (int, np.int, np.int8, np.int16, np.int32, np.int64, np.uint, np.uint8, np.uint16, np.uint32, np.uint64)
float_type = (float, np.float, np.float16, np.float32, np.float64)
```
Now if anywhere in the qcodes code base we need to see if an object is an integer: 

```python
from qcodes.math.datatypes import int_type

if isinstance(a, int_type): 
  # something usefull
```